src/components/dapp-staking/RegisterDapp.vue

```diff
@@ -0,0 +1,332 @@
+    const store = useStore();
+    const theme = computed<string>(() => store.getters['general/theme']);
+    const isDark = ref<boolean>(theme.value.toLowerCase() === 'dark');
+    const data = reactive<NewDappItem>({ tags: [] } as unknown as NewDappItem);
+    const isModalAddDeveloper = ref<boolean>(false);
+    const currentDeveloper = ref<Developer>(initDeveloper());
+    const currentCategory = ref<LabelValuePair>(possibleCategories[0]);
+    const dappForm = ref();
+    const errors = ref({
+      builders: '',
+      community: '',
+      platform: '',
+    });
+
+    // make a placeholder for add logo
+    data.icon = new File([], 'Add a logo image'); // TODO translate
+    data.developers = [];
+    data.communities = [];
+    data.platforms = [];
+    data.tags = [];
+    data.mainCategory = currentCategory.value.value as Category;
+    data.license = possibleLicenses[0].value;
+    data.iconFile = '';
+    data.address = '0x00000....';
+
+    data.images = [];
+    data.images.push(new File([], 'Add an image')); // Add image placeholder
+    data.imagesContent = [];
+    data.imagesContent.push('');
+
+    const isValidAddress = (address: string): boolean => isEthereumAddress(address); // || isValidAddressPolkadotAddress(address);
+    // TODO uncoment the code above when we will support ink contract.
+
+    const updateDappLogo = (): void => {
+      const reader = new FileReader();
+      reader.readAsDataURL(data.icon);
+      reader.onload = () => {
+        data.iconFile = reader.result?.toString() || '';
+      };
+      reader.onerror = (error) => console.error(error);
+    };
+
+    const handleDappChanged = (newData: NewDappItem): void => {
+      newData.descriptionMarkdown = newData.description ? sanitizeData(newData.description) : '';
+
+      currentCategory.value =
+        possibleCategories.find((x: LabelValuePair) => x.value === data.mainCategory) ??
+        possibleCategories[0];
+      data.ref = newData;
+      console.log(newData.ref);
+    };
+
+    const validateCustomComponents = (): boolean => {
+      errors.value.builders =
+        data.developers.length > 1 ? '' : t('dappStaking.modals.builder.error.developersRequired');
+      errors.value.community =
+        data.communities.length > 0 ? '' : t('dappStaking.modals.community.communityRequired');
+      errors.value.platform = data.platforms.length > 0 ? '' : 'Platform missing';
+      console.log(errors.value);
+
+      for (const [key, value] of Object.entries(errors.value)) {
+        if (value) {
+          return false;
+        }
+      }
+
+      return true;
+    };
+
+    const handleSubmit = (): void => {
+      dappForm?.value?.validate().then(async (success: boolean) => {
+        if (success && validateCustomComponents()) {
+          alert('Submitted');
+        }
+      });
+    };
+
+    watch([theme], (val) => {
+      isDark.value = val[0].toLowerCase() === 'dark';
+    });
+
+    return {
+      data,
+      isDark,
```

Comment:
`isDark` is not used in the `template` in this file

Reply:
> `isDark` is not used in the `template` in this file

Removed as not needed in [this](https://github.com/AstarNetwork/astar-apps/pull/522/commits/b5d06c46c916a0a8ba47c68485e871194e612ee2) commit